pkg/kubelet/pod/pod_manager.go

```diff
@@ -335,11 +352,28 @@ func (pm *basicManager) getOrphanedMirrorPodNames() []string {
 	return podFullNames
 }
 
-func (pm *basicManager) DeleteOrphanedMirrorPods() {
-	podFullNames := pm.getOrphanedMirrorPodNames()
-	for _, podFullName := range podFullNames {
-		pm.MirrorClient.DeleteMirrorPod(podFullName, nil)
+func (pm *basicManager) GetPendingPodUID(podFullName string) (kubetypes.UIDWithGracePeriod, bool) {
+	pm.lock.RLock()
+	defer pm.lock.RUnlock()
+	uid, ok := pm.pendingPodByFullName[podFullName]
+	return uid, ok
+}
+
+func (pm *basicManager) IsPodPendingExit(uid types.UID) bool {
+	pm.lock.RLock()
+	defer pm.lock.RUnlock()
+	for _, uidWithGracePeriod := range pm.pendingPodByFullName {
+		if uidWithGracePeriod.UID == kubetypes.ResolvedPodUID(uid) {
+			return true
+		}
 	}
+	return false
+}
+
+func (pm *basicManager) DeletePendingPod(podFullName string) {
+	pm.lock.Lock()
+	defer pm.lock.Unlock()
+	delete(pm.pendingPodByFullName, podFullName)
```

Comment:
this is only place i see delete happening, but its only caller is when handling mirror pods?

Reply:
Population of the map is narrowed in DeletePod().